### PR TITLE
Add stylesheet

### PR DESCRIPTION
**Pull request by svenanderzen**
## Description
Adds a CSS file for styling the UI. 

## Important files to review
`main.py`
`ui/styles.css`

## See also
Se #88.

### CODE REVIEW

**Comment by svenanderzen** [comment]
Perhaps you can start putting in the CSS stuff in this branch when you feel ready @jaokob ? 👍

File: main.py [hunk 1]
```diff
@@ -41,13 +41,21 @@ def __init__(self):
 
     def setup_ui(self):
         """Loads the main.ui file and sets up the window and grid."""
+
```
Comment (mictab) [inline main.py]: Why so much space here?

File: ui/styles.qss [hunk 2]
```diff
@@ -0,0 +1,218 @@
+QToolTip
+{
+    border: 1px solid #76797C;
+    background-color: rgb(90, 102, 117);;
```
Comment (mictab) [inline ui/styles.qss]: Remove double semi-colon.

File: ui/styles.qss [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    color: white;
+    padding: 5px;
+    opacity: 200;
+}
+ /* QInputDialog styling */
+QWidget
+{
+    color: #eff0f1;
+    background-color: #262730;
+    selection-background-color:#3daee9;
```
Comment (mictab) [inline ui/styles.qss]: Add space after colon.

File: ui/styles.qss [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    background-color: #3daee9;
+    padding-top: -15px;
+    padding-bottom: -17px;
+}
+
+QPushButton:checked{
+    background-color: #76797C;
+    border-color: #6A6969;
+}
+/*
```
Comment (mictab) [inline ui/styles.qss]: Why is this commented out?

File: ui/styles.qss [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+}
+
+QMenu {
+    background-color: #262730;
+    border: 1px solid black;
+}
+
+QMenu::item {
+	padding: 2px 25px 2px 20px;
+	color: #eff0f1;
```
Comment (mictab) [inline ui/styles.qss]: Mixing tabs and spaces here.

File: ui/styles.qss [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+}
+QMenu::item:selected {
+    border-color: darkblue;
+    background: rgba(100, 100, 100, 150);
+}
+
+QPushButton::menu-indicator  {
+    subcontrol-origin: padding;
+    subcontrol-position: bottom right;
```
Comment (mictab) [inline ui/styles.qss]: Sure this shouldn't be bottom-right? :)

File: ui/styles.qss (hunk 4 above)
Comment (jaokob) [inline ui/styles.qss]: hooverbug

Sometimes, on mouse hover the menu bar gets this bluish/cyan outline, I thought that it was that snippet of qss which caused it. It wasn't, and for some reason I forgot to undo the comment. I'm still trying to figure out why this happens.

File: ui/styles.qss (hunk 6 above)
Comment (jaokob) [inline ui/styles.qss]: I mean, I'm not 100% sure. But according to http://doc.qt.io/qt-4.8/stylesheet-examples.html#customizing-qpushbutton the syntax is hyphen-less.

Comment (mictab) [inline ui/styles.qss]: Looks right then! :)

**Comment by mictab** [comment]
How's this going?

**Comment by jaokob** [comment]
I'm waiting for the buttons to get added, otherwise it is ready for merge I reckon.

**Comment by jaokob** [comment]
I'd like some feedback on the icons and CSS if possible. Also if someone could help me make the volume slide look good that would be awesome. Currently the transparent background is not working as intended, and I'm not sure on how to fix it.

File: ui/frame.ui [hunk 7]
```diff
@@ -41,12 +41,15 @@
...
-     <width>100</width>
-     <height>23</height>
+     <width>50</width>
+     <height>50</height>
     </rect>
    </property>
+   <property name="accessibleName">
+    <string>pauseBtn</string>
+   </property>
    <property name="text">
```
Comment (kaszim) [inline ui/frame.ui]: Cant this property simply be removed if it is empty? Or is the default value a non-empty string?

File: ui/frame.ui [hunk 8]
```diff
@@ -57,12 +60,15 @@
...
-     <width>100</width>
-     <height>23</height>
+     <width>50</width>
+     <height>50</height>
     </rect>
    </property>
+   <property name="accessibleName">
+    <string>deleteBtn</string>
+   </property>
    <property name="text">
```
Comment (kaszim) [inline ui/frame.ui]: See 👆

File: ui/rewoundframe.ui [hunk 9]
```diff
@@ -41,12 +44,15 @@
...
-     <width>100</width>
-     <height>23</height>
+     <width>50</width>
+     <height>50</height>
     </rect>
    </property>
+   <property name="accessibleName">
+    <string>pauseBtn</string>
+   </property>
    <property name="text">
```
Comment (kaszim) [inline ui/rewoundframe.ui]: See 👆

File: ui/rewoundframe.ui [hunk 10]
```diff
@@ -57,12 +63,15 @@
...
-     <width>100</width>
-     <height>23</height>
+     <width>50</width>
+     <height>50</height>
     </rect>
    </property>
+   <property name="accessibleName">
+    <string>forwBtn</string>
+   </property>
    <property name="text">
```
Comment (kaszim) [inline ui/rewoundframe.ui]: 👆

File: ui/rewoundframe.ui [hunk 11]
```diff
@@ -73,12 +82,15 @@
...
-     <width>100</width>
-     <height>23</height>
+     <width>50</width>
+     <height>50</height>
     </rect>
    </property>
+   <property name="accessibleName">
+    <string>backBtn</string>
+   </property>
    <property name="text">
```
Comment (kaszim) [inline ui/rewoundframe.ui]: 👆

File: videoframes.py [hunk 12]
```diff
@@ -107,8 +108,10 @@ def toggle_playback(self):
         """Toggles playback of a stream."""
         if self.player.is_playing():
             self.player.pause()
+            self.toggle_button()
```
Comment (jaokob) [inline videoframes.py]: True that 👍

File: ui/styles.qss [hunk 13]
```diff
@@ -0,0 +1,217 @@
...
+    padding-bottom: 5px;
+    padding-left: 10px;
+    padding-right: 10px;
+    border-radius: 2px;
+    color: #454545;
+}
+
+QPushButton[accessibleName="deleteBtn"]
+{
+background-color: transparent;
```
Comment (kaszim) [inline ui/styles.qss]: No indenting?

File: videoframes.py [hunk 14]
```diff
@@ -131,6 +134,25 @@ def toggle_select(self):
         else:
             self.select()
 
+    def toggle_button(self):
+        """Toggles the icon of a play/pause button"""
+        if self.player.is_playing():
+            self.setStyleSheet(
+                'QPushButton[accessibleName="pauseBtn"] {' +
```
Comment (kaszim) [inline videoframes.py]: Should probably be put in constants.py.

File: ui/frame.ui (hunk 7 above)
Comment (jaokob) [inline ui/frame.ui]: Explain, I use that property to skin certain things. I.e. the pause button in this case. 

`QPushButton[accessibleName="pauseBtn"]:focus` 

for example

File: videoframes.py (hunk 14 above)
Comment (jaokob) [inline videoframes.py]: True, I'll fix it.

File: ui/styles.qss (hunk 13 above)
Comment (jaokob) [inline ui/styles.qss]: I'll fix this

File: ui/frame.ui (hunk 7 above)
Comment (kaszim) [inline ui/frame.ui]: I mean the following line: `<property name="text">`

Comment (jaokob) [inline ui/frame.ui]: Oh, I don't know. This is created by qt-creator when I set an objects accessibleName.

Comment (kaszim) [inline ui/frame.ui]: Can you test without them? 😃

Comment (jaokob) [inline ui/frame.ui]: changed my qss, so I could completely remove it.

**Comment by kaszim** [comment]
I found a bug. If one switches position of buttons then pause button loses it's styling.

File: videoframes.py [hunk 15]
```diff
@@ -106,12 +106,11 @@ def context_menu(self, event):
 
     def toggle_playback(self):
         """Toggles playback of a stream."""
+        self.toggle_button()
```
Comment (mictab) [inline videoframes.py]: Can we do this after the conditionals? Just a nitpick 😅 Doesn't really matter though, up to you.

Comment (jaokob) [inline videoframes.py]: Sure thing!

**Comment by jaokob** [comment]
@kaszim could you tell me how to reproduce? Seems strange since as long as the object retains the same name, it should work.

File: main.py (hunk 1 above)
Comment (jaokob) [inline main.py]: This should be fixed at the moment.

**Pull request merged**